### PR TITLE
Cypress v12.8.1 - Fixes for panels/Toast message fix

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
[Describe what this change achieves]
Fixes for cypress testing of panels (with each test creating its own data and cleaning up after itself).
Fixed toast message not appearing on visualization being added to dashboard.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TackAdam** [comment]
Screenshot 2023-06-27 at 4 38 11 PM
All 29 test reliably passing.

File: .cypress/integration/3_panels.spec.ts [hunk 1]
```diff
@@ -10,578 +10,624 @@ import {
...
     cy.get('div[data-test-subj="sampleDataSetCardflights"]')
       .contains(/(Add|View) data/)
       .trigger('mouseover')
       .click();
-    cy.wait(delay * 3);
   });
-});
 
-describe('Creating visualizations', () => {
   beforeEach(() => {
```
Comment (TackAdam) [inline .cypress/integration/3_panels.spec.ts:30]: Each test will now delete all the saved dashboards and visuals to "cleanup" before executing its testable metric.

File: public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout_so.tsx [hunk 2]
```diff
@@ -173,14 +173,17 @@ export const VisaulizationFlyoutSO = ({
...
-          onSuccess: `Visualization ${newVisualizationTitle} successfully added!`,
-          onFailure: `Error in adding ${newVisualizationTitle} visualization to the panel`
-        });
+      const visualizationsWithNewPanel = addVisualizationPanel({
+        savedVisualizationId: selectValue,
+        onSuccess: `Visualization ${newVisualizationTitle} successfully added!`,
+        onFailure: `Error in adding ${newVisualizationTitle} visualization to the panel`,
+      });
     }
+    setToast(`Visualization ${newVisualizationTitle} successfully added!`);
```
Comment (TackAdam) [inline public/components/custom_panels/panel_modules/visualization_flyout/visualization_flyout_so.tsx:186]: Toast message for a visual being added to a dashboard was not ever appearing.

File: .cypress/integration/3_panels.spec.ts [hunk 3]
```diff
@@ -10,578 +10,624 @@ import {
...
   });
 
-  describe('with a SavedObjects Panel', () => {
+  describe('Testing a panel', () => {
     beforeEach(() => {
-      createSavedObjectPanel();
-      moveToPanelHome();
-      cy.get('.euiTableRow').should('have.length', 1);
+      const test_name = `test_${new Date().getTime()}`;
+      createSavedObjectPanel(test_name).as('thePanel');
```
Comment (TackAdam) [inline .cypress/integration/3_panels.spec.ts:253]: Each tested feature of a panel should be independent and exclusive to the intended functionality.

File: .cypress/integration/3_panels.spec.ts [hunk 4]
```diff
@@ -10,578 +10,624 @@ import {
...
+      createVisualization(
+        PPL_VISUALIZATIONS_NAMES[1],
+        PPL_VISUALIZATIONS[1],
+        PPL_VISUALIZATION_CONFIGS[1]
+      ).as('vis1');
+
+      cy.then(function () {
+        addVisualizationsToPanel(this.thePanel, [this.vis1.id]);
+        moveToThePanel(this.thePanel.id);
+        cy.get('[data-test-subj="breadcrumb"]').click({ force: true });
```
Comment (TackAdam) [inline .cypress/integration/3_panels.spec.ts:391]: When adding a visualization to a panel created from code, forcing a refresh ensures that cypress doesn't execute commands before the visual has appeared on the page.

File: .cypress/integration/3_panels.spec.ts [hunk 5]
```diff
@@ -10,578 +10,624 @@ import {
...
 
-  describe('with a SavedObjects Panel', () => {
+  describe('Testing a panel', () => {
     beforeEach(() => {
-      createSavedObjectPanel();
-      moveToPanelHome();
-      cy.get('.euiTableRow').should('have.length', 1);
+      const test_name = `test_${new Date().getTime()}`;
+      createSavedObjectPanel(test_name).as('thePanel');
+      cy.then(function (){
```
Comment (TackAdam) [inline .cypress/integration/3_panels.spec.ts:254]: Creating objects using .as('object') adds the variable to cypress.
We are then able to access these objects using cy.then(function (){this.});

**Review by ps48 — APPROVED** [review]

**Review by pjfitzgibbons — APPROVED** [review]

**Pull request merged**